common/src/main/java/com/spotify/spydra/util/StorageUtils.java

```diff
@@ -0,0 +1,82 @@
+package com.spotify.spydra.util;
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.util.Iterator;
+
+
+public class StorageUtils {
+    private static final Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    public Bucket checkBucket(String bucketName) {
+        if (bucketName == null) {
+            throw new IllegalArgumentException(
+                    "Provide bucket name.");
+        } else {
+            return storage.get(bucketName, Storage.BucketGetOption.fields());
+        }
+    }
+
+    public Bucket createBucket(String bucketName) {
+        Bucket bucket = storage.create(BucketInfo.newBuilder(bucketName)
+                .setLocation("europe-west1")
+                .build());
+        return bucket;
+    }
+
+    public Blob createBlob(String bucketName, String blobName) {
+        BlobId blobId = BlobId.of(bucketName, blobName);
+        BlobInfo blobInfo = BlobInfo.newBuilder(blobId).setContentType("text/plain").build();
+        Blob blob = storage.create(blobInfo);
+        return blob;
+    }
+
+    public Page<Blob> listBucket(String bucketName, String directory) {
+        if (bucketName == null) {
+            throw new IllegalArgumentException(
+                    "Provide bucket name.");
+        } else {
+            Bucket bucket = storage.get(bucketName);
+            if (bucket == null) {
+                throw new IllegalArgumentException(
+                        "No such bucket.");
+            }
+            Page<Blob> blobs = bucket.list(Storage.BlobListOption.prefix(directory));
+            return blobs;
+        }
+    }
+
+    public int getFileCount(String bucketName, String directory) {
+        Iterator bucketIterator = listBucket(bucketName, directory).getValues().iterator();
```

Comment:
Notice that you are counting here the objects on first page, not overall. Compare iterateAll() with getValues()

Reply:
I also remember you wondering yourself whether we should do full object count just to decide whether target exists (!= 0). I think we should improve here, but don't know top of my head how.